examples/bits.rs

```diff
@@ -1,98 +1,249 @@
+    // We also expect to be drawing a row for each byte in the text, with an
+    // additional separator row for each *character*, half above and half below
+    // the center of the screen. This computes how many rows of text we will
+    // draw, then moves the turtle appropriately.
+    let byte_rows = TEXT.len();
+    let char_gaps = TEXT.chars().count();
+    let top_edge = BIT_HEIGHT * ((byte_rows + char_gaps) as f64 / 2.0);
+    // The turtle starts from the top right of the region,
+    turtle.forward(top_edge);
+    turtle.right(90.0);
+    turtle.forward(right_edge);
+    // and walks left
+    turtle.left(180.0);
+
+    draw_text(&mut turtle, TEXT);
+
+    // The `draw_number` function reads bits from left to right, so the turtle
+    // should also walk from left to right. The `draw_number` function expects
+    // that it will be drawing rows sixteen bits long, so it needs to move
+    // forward another four bits' worth of space in order to be in the correct
+    // spot.
+    turtle.forward(8.0 * BIT_BOX / 2.0);
+    turtle.forward(16.0 * BIT_BOX / 2.0);
+    // Then, it needs to turn around, to walk in the other direction.
+    turtle.right(180.0);
 
-/// The number of bits in a single element of the bytes Vec in BitVec
-/// Set to 8 because BitVec stores u8 values. If we stored u16, we would use 16 here
-const BITS: usize = 8;
-
-/// An implementation of a vector of bits that can be easily set individually
-/// Note that this is meant to be a teaching implementation and so is not necessarily the most
-/// sophisticated or performant way to do everything. The goal is to make the code understandable
-/// and correct.
-/// See the bit-vec crate and others on crates.io for more advanced implementations.
-#[derive(Debug, Clone, Default)]
-pub struct BitVec {
-    bytes: Vec<u8>,
+    draw_number(&mut turtle, NUMBER);
+
+    // Reader exercise:
+    //
+    // The IEEE-754 format for `f64` numbers separates them into three parts:
+    //
+    // 1. The sign marks whether the number is positive or negative: 1 bit
+    // 2. The exponent marks how far from zero the number is: 11 bits
+    // 3. The fraction describes the number: 52 bits.
+    //
+    // Using these widths (1 bit, 11 bits, 52 bits), the knowledge that
+    // `&BitSlice` is a normal Rust slice, and the API documentation for
+    // `std::iter::Iterator`, see if you can display each portion of an `f64`
+    // as its own row.
+    //
+    // Hints:
+    //
+    // - The variable `bits` is set up to view the entire number, from most
+    // significant bit to least.
+    // - You can get access to a structure that performs iteration by calling
+    //   `bits.iter()`.
+    // - You can use the `Iterator::by_ref` method to prevent `Iterator` adapter
+    //   functions from destroying the source iterator.
+    // - `&BitSlice` is an ordinary Rust slice, so you can use `[start .. end]`
+    //   range indexing to get smaller pieces of it.
```

Comment:
(optional) nit: This exercise is probably best placed at the bottom of `draw_number`. Otherwise, the context of "this is an extension of the ideas in `draw_number`" is lost. You could also refer to `draw_number` in the exercise text. Either works to get the reader to understand where to start with this and where they can look if they get stuck.

Reply:
Yeah that's a good call.